tests/CMakeLists.txt

```diff
@@ -53,6 +54,16 @@ target_compile_options(gsl_tests_config INTERFACE
     >
 )
 
+target_link_libraries(gsl_tests_config INTERFACE
+    $<$<CXX_COMPILER_ID:GNU>:
+        -pthread
```

Comment:
CMake provides functionality to discover currently used threading library.

`find_package(Threads)`
and
`${CMAKE_THREAD_LIBS_INIT}` or `Threads::Threads`

https://cmake.org/cmake/help/v3.9/module/FindThreads.html